### ISSUE TITLE
Mako module stopped work

### ISSUE THREAD

**Issue by timsofteng** [nix-community/home-manager#2028]
Hello. I find `mako` notification module stopped work. Daemon doesn't start. I can't see it in my processes and notifications doen't work.
Is it knowing issue?

**Comment by sumnerevans** [comment]
I have the following in my config to fix the issues with dbus: https://git.sr.ht/~sumner/home-manager-config/tree/master/item/modules/window-manager/wayland.nix#L64

**Comment by berbiche** [comment]
This might be of interest: https://github.com/alebastr/sway-systemd

**Comment by asymmetric** [comment]
Can we reopen this issue? As it is, the `mako` module is broken.

@sumnerevans could you point a permalink to the relevant portion of your config?

**Comment by berbiche** [comment]
@asymmetric hi, do you have multiple notification daemons on your system? They are likely fighting to register for the notification daemon.

If you run the following command, you should get whether a notification daemon (mako or something else) is receiving notifications: `dbus-send --print-reply --dest=org.freedesktop.DBus /org/freekdesktop/DBus org.freedesktop.DBus.ListNames | grep Notifications`

**Comment by asymmetric** [comment]
@berbiche I don't have any notification daemon running, actually. Running that snippet returns empty.

To fix the issue, I now run mako explicitly in my sway config with `exec mako`, but this wasn't necessary before.

Can you instead confirm that just having `programs.mako.enable = true` in your config gives a different result?

**Comment by sumnerevans** [comment]
Not sure if this actually fixes people's problems, but I see this in my git commit log, which means that it maybe is necessary? https://github.com/sumnerevans/home-manager-config/commit/f82aacb32cb25145b614bc1d4fd907617c416b6e#diff-60d8bd9845e95cde122a9132e44f5da4e55231a3505d10548de91e751aaa2afa

**Comment by samhh** [comment]
Is the mako module as currently defined supposed to start it as a daemon? I've had to define my own systemd unit instead:

```nix
{
  systemd.user.services.mako = {
    Install.WantedBy = [ "sway-session.target" ];
    Service.ExecStart = "${pkgs.mako}/bin/mako";
  };
}
```

**Comment by benjumanji** [comment]
Every single solution here seems to miss the forest for the trees.

```
  xdg.dataFile."dbus-1/services/fr.emersion.mako.service" = {
    enable = config.services.mako.enable;
    source = "${pkgs.mako}/share/dbus-1/services/fr.emersion.mako.service";
  };
```

Is sufficient to have mako work without needing any kind of manual spawning or having to write an extra systemd service.  Of course that begs the question: why isn't there a `services.dbus.packages` or some option to mass link dbus activatable services into `XDG_DATA_HOME` ?

**Comment by berbiche** [comment]
@benjumanji Absolutely, IIRC there's an option to specify the paths to install in the HM closure and systemd services are there, maybe dbus services should also be there?
I believe the NixOS setting is `environment.pathsToLink`, but I can't find the HM equivalent.

<i><sub>n.b. I haven't touched my Nix config in months nor kept up with HM development</sub></i>

**Comment by Rosuavio** [comment]
I could not agree more @benjumanji, @berbiche! I made https://github.com/nix-community/home-manager/pull/7064 . Its not just like `environment.pathsToLink`, but I like that the user dose not have to add the package to `home.packages` as well (as would be the case with a `pathsToLink` like solution).

Made a mako fix using it https://github.com/nix-community/home-manager/pull/7065

**Comment by benjumanji** [comment]
@Rosuavio I completely forgot I made this comment, upgraded hm today to 25.05 and was _pleasantly surprised_ to find a file conflict. Great work! Thank you!
